pygmt/io.py

```diff
@@ -2,10 +2,14 @@
 PyGMT input/output (I/O) utilities.
 """
 
+from typing import Literal
```

Comment:
Since we're going to deprecate `load_dataarray`(xref: https://github.com/GenericMappingTools/pygmt/pull/3919#issuecomment-2812018012), we should revert the changes in this file.

Reply:
Do we want to change the default engine in load_dataarray to `engine="gmt"` or no?

Reply:
Changing the default engine to `engine="gmt"` means that users have to set `decode_kind`, so it's a breaking change. It makes little sense to introduce a breaking change to a deprecated function, so my answer is no.

Reply:
Ok, I've reverted changes to `io.py` in commit f722b479903458f4349a3e0afb75d6c6fcae461f. So keeping default engine behaviour as before for the `load_dataarray` function that will be deprecated in #3922.